### PR TITLE
WIP - Transactions purchases

### PR DESCRIPTION
**Pull request by rmisio**
This PR brings in the functionality for the Purchases tab on the Transactions page.

Notes:
- This should be run in conjunction with this [server PR](https://github.com/OpenBazaar/openbazaar-go/pull/515), although that PR is likely in flux. When that PR changes, this PR will need to be adjusted.
- I've hidden the sort icon on the Date column since it looks like we'll be moving to a Sort drop down.
- The listing column is not clickable at this moment. I'm waiting for some back-end data, after which it will link to the listing detail page.

### CODE REVIEW

File: js/collections/Transactions.js [hunk 1]
```diff
@@ -1,22 +1,83 @@
...
+      title: 'try again',
+      total: 156606,
+      unreadChatMessages: 0,
+      vendorHandle: '',
+      vendorId: 'QmbSYKcypVVZNVQ4tqgG6oE8kNBapjgoU5676fBp62hcWE',
+    };
+  }
+
+  // temporary to test pagination with dummy data
+  fetch2() {
```
Comment (rmisio) [inline js/collections/Transactions.js]: I'm leaving this in because it may be useful for me to test pagination in the other Transaction tabs.

**Comment by morebrownies** [comment]
Hey @rmisio how do I go about getting transactions onto my node? Trying to test this PR.

**Comment by rmisio** [comment]
> Hey @rmisio how do I go about getting transactions onto my node? Trying to test this PR.

Hmm... I think the easiest way would be to use [Josh's PR](https://github.com/OpenBazaar/openbazaar-desktop/pull/470) to create some (you click Buy Now and then Purchase). Then once you have some, you could switch back over to this PR.

**Comment by jjeffryes** [comment]
@morebrownies use #472 now, I closed the earlier one. This one has code for purchasing that is more complete.

**Comment by morebrownies** [comment]
Thanks will give it a shot in a few

File: js/views/transactions/table/Table.js [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+    this.indexedViews = {};
+    this.views.forEach(view => {
+      if (this.type === 'purchases') {
+        this.indexedViews[view.model.get('vendorId')] = view;
+      }
+    });
+  }
+
+  get transactionsPerPage() {
+    return 5;
```
Comment (rmisio) [inline js/views/transactions/table/Table.js]: For testing I have this quite low, but I plan to raise it later... probably 20.

**Comment by rmisio** [comment]
@jjeffryes I'm going to have Sales ready in the next day or two. I'll just go ahead and close this one for now, since the code changed a good amount (I refactored the Purchases functionality to be more abstract and re-usable on the other tabs).

**Pull request closed without merging**